src/utils/relative_point.rs

```diff
@@ -0,0 +1,31 @@
+#[cfg(feature = "wayland_frontend")]
+use crate::wayland::seat::WaylandFocus;
+
+use super::{Logical, Point};
+
+/// Represents a point on a surface, relative to that surface's origin in
+/// global compositor space.
+#[derive(Debug, Clone)]
+pub struct RelativePoint<S> {
+    /// The surface that the pointer is over.
+    pub surface: S,
+    /// The relative location of the cursor within the surface.
+    pub loc: Point<f64, Logical>,
+}
+
+#[cfg(feature = "wayland_frontend")]
+impl<S: WaylandFocus> RelativePoint<S> {
+    /// Calculates the relative position of a cursor on a target surface,
+    /// given a point and that  surface's origin coordinates, both in global
+    /// compositor space.
+    pub fn on_focused_surface(
+        point: Point<f64, Logical>,
+        focus: S,
+        focus_origin: Point<i32, Logical>,
```

Comment:
Please use `impl Into<Point<...>>` as arguments whenever possibly to allow for example a tuble to be passed to these functions directly instead of just `Point`s.

Reply:
Done in 180b08e1126884846f1c50e9b610babad110aec8.